apps/server/src/modules/fwu-learning-contents/uc/fwu-learning-contents.uc.ts

```diff
@@ -0,0 +1,50 @@
+import { Injectable, NotFoundException, InternalServerErrorException, Inject } from '@nestjs/common';
+import { GetObjectCommand, S3Client } from '@aws-sdk/client-s3';
+import { Readable } from 'stream';
+import { Logger } from '@src/core/logger';
+import { S3Config } from '../interface/config';
+
+@Injectable()
+export class FwuLearningContentsUc {
+	constructor(
+		private logger: Logger,
+		@Inject('S3_Client') readonly client: S3Client,
+		@Inject('S3_Config') readonly config: S3Config
+	) {
+		this.logger.setContext(FwuLearningContentsUc.name);
+	}
+
+	private async getResponse(path: string) {
+		const request = new GetObjectCommand({
+			Bucket: this.config.bucket,
+			Key: path.toString(),
+		});
+		const response = await this.client.send(request).catch((error: unknown) => {
+			if (error && typeof error === 'object' && 'name' in error && 'stack' in error) {
+				if (error.name === 'NoSuchKey') {
+					throw new NotFoundException({ cause: error.stack });
+				} else {
+					throw new InternalServerErrorException({ name: error.name, cause: error.stack });
+				}
+			}
+			throw new InternalServerErrorException({ name: undefined, cause: error });
```

Comment:
always have a message for your exceptions, makes finding the source even easier and one can search for in logs!

Reply:
@Metauriel here is the updated version

Reply:
bitte trotzdem den Kommentar von Loki-Afro beachten. und wenn es nur ist name: "unexpected error on reading file from FWU S3 storage".

Das hilft zumindest die code stelle zu finden wenn der fehler im log auftaucht.